tests/ui/fmt/help-message-issue-129966-reorder-remove-format-specifier.rs

```diff
@@ -0,0 +1,14 @@
+// Test checks for the help messages in the error output/
+//@ check-run-results
+
+struct Foo(u8, u8);
+
+fn main() {
+    let f = Foo(1, 2);
+    format!("{f:?#}");
+//~^ ERROR invalid format string: unknown format identifier '?#'
```

Comment:
Problem [REV-RUSTFIX 4/4]: since we would like the help message to be present even in absence of tools like rustfix, we should also match help messages:

```rs
    let _ = format!("{f:?x}");
    //~^ ERROR invalid format string: unknown format identifier '?x'
    //~| HELP did you mean to write `x?`?
```